### PR TITLE
Bug with closing tabs in processes panel

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
Fixes bug when closing tabs in processes panel.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/8479
https://github.com/eclipse/che/issues/8272

### CODE REVIEW

**Comment by vitaliy-guliy** [comment]
ci-test

**Review by ashumilova — APPROVED** [review]

File: ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java [hunk 1]
```diff
@@ -268,16 +268,33 @@ public void remove() {
...
       tabs2Widgets.remove(tab);
 
       // remove item from drop-down menu
       final MenuItemWidget listItemWidget = widgets2ListItems.remove(widget);
       if (listItemWidget != null) {
         menu.removeListItem(listItemWidget);
       }
+
+      if (tab == selectedTab) {
+        if (tabsPanel.getWidgetCount() > 1) {
```
Comment (vparfonov) [inline ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java]: Here we can replace, I think 
```
 if (tab == selectedTab) {
        if (tabsPanel.getWidgetCount() > 1) {
```
> 
`if (tab == selectedTab && tabsPanel.getWidgetCount() > 1) {`

Comment (vitaliy-guliy) [reply ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java]: Done. Thanks

**Review by vparfonov — APPROVED** [review]

File: ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java [hunk 2]
```diff
@@ -268,16 +268,31 @@ public void remove() {
...
+        Widget widgetToSelect;
+        if (removedTabIndex < tabsPanel.getWidgetCount() - 1) {
+          widgetToSelect = tabsPanel.getWidget(removedTabIndex);
+        } else {
+          widgetToSelect = tabsPanel.getWidget(tabsPanel.getWidgetCount() - 2);
+        }
+
+        if (widgetToSelect instanceof Tab) {
+          selectTab((Tab) widgetToSelect);
+          onTabClicked((Tab) widgetToSelect);
```
Comment (vparfonov) [inline ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java:293]: Do we need onTabClicked() here? Look strange for me

File: ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java [hunk 3]
```diff
@@ -268,16 +268,31 @@ public void remove() {
...
+
+      if (tab == selectedTab && tabsPanel.getWidgetCount() > 1) {
+        Widget widgetToSelect;
+        if (removedTabIndex < tabsPanel.getWidgetCount() - 1) {
+          widgetToSelect = tabsPanel.getWidget(removedTabIndex);
+        } else {
+          widgetToSelect = tabsPanel.getWidget(tabsPanel.getWidgetCount() - 2);
+        }
+
+        if (widgetToSelect instanceof Tab) {
```
Comment (vparfonov) [inline ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java:291]: Are you sure `widgetToSelect` can be not a Tab?

Comment (vitaliy-guliy) [reply ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java:291]: Tabs panel may contain different widgets. Tabs, menu button, etc.

File: ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java (hunk 2 above)
Comment (vitaliy-guliy) [reply ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/panel/SubPanelViewImpl.java:293]: We need some reaction when selecting a tab.

**Review by vparfonov — APPROVED** [review]

**Comment by vitaliy-guliy** [comment]
ci-build

**Comment by vitaliy-guliy** [comment]
ci-build

**Comment by musienko-maxim** [comment]
ci-build

**Review by musienko-maxim — APPROVED** [review]

**Pull request merged**